File: java/src/main/java/sk/epostak/sdk/resources/BoxResource.java
package sk.epostak.sdk.resources;

import sk.epostak.sdk.HttpClient;
import sk.epostak.sdk.models.BoxCreateRequest;
import sk.epostak.sdk.models.BoxListParams;
import sk.epostak.sdk.models.BoxScheduleRequest;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * ePošťák Box durable execution layer for staged, scheduled, and retryable
 * Peppol dispatch.
 */
public final class BoxResource {

    private final HttpClient http;

    public BoxResource(HttpClient http) {
        this.http = http;
    }

    public Map<String, Object> list(BoxListParams params) {
        Map<String, Object> qp = new LinkedHashMap<>();
        if (params != null) {
            qp.put("status", params.status());
            qp.put("direction", params.direction());
            qp.put("limit", params.limit());
            qp.put("offset", params.offset());
        }
        return (Map<String, Object>) http.get("/box/items" + HttpClient.buildQuery(qp), Map.class);
    }

    public Map<String, Object> list() {
        return list(null);
    }

    public Map<String, Object> create(BoxCreateRequest request) {
        return (Map<String, Object>) http.post("/box/items", request, Map.class);
    }

    public Map<String, Object> get(String itemId) {
        return (Map<String, Object>) http.get("/box/items/" + HttpClient.encode(itemId), Map.class);
    }

    public Map<String, Object> schedule(String itemId, BoxScheduleRequest request) {
        return (Map<String, Object>) http.post(
                "/box/items/" + HttpClient.encode(itemId) + "/schedule",
                request,
                Map.class
        );
    }

    public Map<String, Object> sendNow(String itemId) {
        return (Map<String, Object>) http.post(
                "/box/items/" + HttpClient.encode(itemId) + "/send-now",
                Map.of(),
                Map.class
        );
    }

    public Map<String, Object> retry(String itemId) {
        return (Map<String, Object>) http.post(
                "/box/items/" + HttpClient.encode(itemId) + "/retry",
                Map.of(),
                Map.class
        );
    }

    public Map<String, Object> cancel(String itemId) {
        return (Map<String, Object>) http.post(
                "/box/items/" + HttpClient.encode(itemId) + "/cancel",
                Map.of(),
                Map.class
        );
    }
}

File: dotnet/src/EPostak/Resources/BoxResource.cs
using EPostak.Models;

namespace EPostak.Resources;

/// <summary>
/// ePošťák Box durable execution layer for staged, scheduled, and retryable
/// Peppol dispatch.
/// </summary>
public sealed class BoxResource
{
    private readonly HttpRequestor _http;

    internal BoxResource(HttpRequestor http) => _http = http;

    public Task<Dictionary<string, object?>> ListAsync(BoxListParams? @params = null, CancellationToken ct = default)
    {
        var qs = HttpRequestor.BuildQuery(
            ("status", @params?.Status),
            ("direction", @params?.Direction),
            ("limit", @params?.Limit?.ToString()),
            ("offset", @params?.Offset?.ToString()));
        return _http.RequestAsync<Dictionary<string, object?>>(HttpMethod.Get, $"/box/items{qs}", ct);
    }

    public Task<Dictionary<string, object?>> CreateAsync(BoxCreateRequest request, CancellationToken ct = default)
        => _http.RequestAsync<Dictionary<string, object?>>(HttpMethod.Post, "/box/items", request, ct);

    public Task<Dictionary<string, object?>> GetAsync(string itemId, CancellationToken ct = default)
        => _http.RequestAsync<Dictionary<string, object?>>(HttpMethod.Get, $"/box/items/{Uri.EscapeDataString(itemId)}", ct);

    public Task<Dictionary<string, object?>> ScheduleAsync(string itemId, BoxScheduleRequest request, CancellationToken ct = default)
        => _http.RequestAsync<Dictionary<string, object?>>(
            HttpMethod.Post,
            $"/box/items/{Uri.EscapeDataString(itemId)}/schedule",
            request,
            ct);

    public Task<Dictionary<string, object?>> SendNowAsync(string itemId, CancellationToken ct = default)
        => _http.RequestAsync<Dictionary<string, object?>>(
            HttpMethod.Post,
            $"/box/items/{Uri.EscapeDataString(itemId)}/send-now",
            ct);

    public Task<Dictionary<string, object?>> RetryAsync(string itemId, CancellationToken ct = default)
        => _http.RequestAsync<Dictionary<string, object?>>(
            HttpMethod.Post,
            $"/box/items/{Uri.EscapeDataString(itemId)}/retry",
            ct);

    public Task<Dictionary<string, object?>> CancelAsync(string itemId, CancellationToken ct = default)
        => _http.RequestAsync<Dictionary<string, object?>>(
            HttpMethod.Post,
            $"/box/items/{Uri.EscapeDataString(itemId)}/cancel",
            ct);
}
